File: JwtWithOAuth2Test/src/main/java/com/example/JwtWithOAuth2Test/api/repository/user/UserRepository.java
package com.example.JwtWithOAuth2Test.api.repository.user;

import com.example.JwtWithOAuth2Test.api.entity.user.User;
import org.springframework.data.jpa.repository.JpaRepository;

public interface UserRepository extends JpaRepository<User, Long> {
    User findByUserId(String userId);
}

File: kotlinWithSecurityJWT/src/main/kotlin/com/springpratice3370/kotlinWithSecurityJWT/user/UserRepository.kt
package com.springpratice3370.kotlinWithSecurityJWT.user

import org.springframework.data.jpa.repository.JpaRepository

interface UserRepository : JpaRepository<User, Long> {

    fun findByEmailOrPhoneNumberOrNickname(email: String, phoneNumber: String, nickname: String): List<User>

    fun findByEmailOrNickname(email: String, nickname: String): List<User>

    fun findByEmail(email: String): User?

    fun findByPhoneNumber(phoneNumber: String): User?

    fun findByNickname(nickname: String): User?

    fun findByIdIn(ids: Set<Long>): List<User>
}
